plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/Ck.hs

```diff
@@ -98,15 +100,30 @@ type CkM uni fun s =
         (ExceptT (CkEvaluationException uni fun)
             (ST s))
 
-instance AsEvaluationFailure CkUserError where
-    _EvaluationFailure = _EvaluationFailureVia CkEvaluationFailure
-
-instance AsUnliftingError CkUserError where
-    _UnliftingError = _UnliftingErrorVia CkEvaluationFailure
-
 instance Pretty CkUserError where
     pretty CkEvaluationFailure = "The provided Plutus code called 'error'."
 
+builtinErrorToCkEvaluationException
+  :: Maybe (Term TyName Name uni fun ())
+  -> BuiltinError
+  -> CkEvaluationException uni fun
+builtinErrorToCkEvaluationException cause =
```

Comment:
Observation: you're not factoring in throwing an exception here (which arguably makes more sense for exception than for errors!)

Reply:
I intentionally prevented myself from making any semantic changes on error types.

Reply:
I didn't mean to change the `CkEvaluationException` type (I could have, but that wasn't my point). 
I meant this direction of change `builtinErrorToCkEvaluationException` -> `throwBuiltinErrorToCkEvaluationException`.
The PR doesn't follow this path consistently.